tasks/request_from_archive/test/unit_tests/test_request_from_archive.py

```diff
@@ -2070,16 +2157,58 @@ def test_handler_output_json_schema(
         }
 
         # Validate the output is correct
-        _OUTPUT_VALIDATE(result_value)
+        _OUTPUT_VALIDATE(result)
 
         # Check the values of the result less the times since those will never match
-        for granule in result_value["granules"]:
+        for granule in result["granules"]:
             for file in granule[request_from_archive.GRANULE_RECOVER_FILES_KEY]:
                 file.pop(request_from_archive.FILE_REQUEST_TIME_KEY, None)
                 file.pop(request_from_archive.FILE_LAST_UPDATE_KEY, None)
                 file.pop(request_from_archive.FILE_COMPLETION_TIME_KEY, None)
 
-        self.assertEqual(expected_result, result_value)
+        self.assertEqual(expected_result, result)
+
+    @patch("request_from_archive.LOGGER.info")
+    def test_set_optional_event_property(
+        self, mock_logger: MagicMock,
+    ):
+        """
+        Unit test for set_optional_event_property function.
+        """
+        mock_event = create_handler_event()
+        mock_target_path_cursor = {
+            "optionalValues": {
+                "config": {
+                    "defaultRecoveryTypeOverride":
+                        "event.meta.collection.meta.orca.defaultRecoveryTypeOverride",
+                    "defaultBucketOverride":
+                        "event.meta.collection.meta.orca.defaultBucketOverride",
+                    "s3MultipartChunksizeMb":
+                        "event.meta.collection.meta.s3MultipartChunksizeMb",
+                    "asyncOperationId":
+                        "event.cumulus_meta.asyncOperationId"
+                }
+            }
+        }
+        mock_target_path_segments = []
+        request_from_archive.set_optional_event_property(
+            mock_event, mock_target_path_cursor, mock_target_path_segments
+            )
+        self.assertEqual(mock_logger.call_count, 4)
+
+        # reset mock_logger for next test
+        mock_logger.reset_mock()
+        event_key = mock_event["event"]["meta"]["collection"]["meta"]
+        mock_event["event"]["cumulus_meta"]["asyncOperationId"] = "123"
+        event_key["orca"]["defaultRecoveryTypeOverride"] = "Standard"
+        event_key["orca"]["defaultBucketOverride"] = "test-bucket"
+        event_key["s3MultipartChunksizeMb"] = 100
+
+        request_from_archive.set_optional_event_property(
+            mock_event, mock_target_path_cursor, mock_target_path_segments
+            )
+        # test that logger is not called since all keys are present in input
+        self.assertEqual(mock_logger.call_count, 0)
```

Comment:
Relying on logger like this isn't ideal, especially since there is no validation on what is being logged. At minimum, check one property that is given a non-null value, and one that does not exist in `event` and is thus given a null value.

Reply:
added additional validation to check output